diff --git a/server/sinatra/server.rb b/server/sinatra/server.rb
index 5237d06..d40f733 100644
--- a/server/sinatra/server.rb
+++ b/server/sinatra/server.rb
@@ -1,6 +1,7 @@
 require 'rubygems'
 require 'bundler'
 require 'pathname'
+require 'rack/session/memcache'
 Bundler.require
 
 $LOAD_PATH.unshift(File.dirname(__FILE__))
@@ -23,7 +24,8 @@ class Controller < Sinatra::Base
   set :views , File.join(SINATRA_ROOT, "views")
   set :haml, :format => :html5
   set :versions, `git log -10 --oneline` || "no git log"
-  enable :sessions
+  #enable :sessions
+  use Rack::Session::Memcache
 
   Store.set ENV['STORE_TYPE'], APP_ROOT
 
