diff --git a/Gemfile b/Gemfile
index 51700b9..c4a2345 100644
--- a/Gemfile
+++ b/Gemfile
@@ -11,6 +11,7 @@ gem "png"
 gem "rest-client"
 gem "ruby-openid"
 gem "couchrest"
+gem "memcache-client", :require => 'memcache'
 
 group :development do
   gem 'ruby-debug', :require => 'ruby-debug', :platform => :mri_18
diff --git a/Gemfile.lock b/Gemfile.lock
index f2f66f0..b573c93 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -41,6 +41,7 @@ GEM
       rbx-require-relative (> 0.0.4)
     linecache19 (0.5.12)
       ruby_core_source (>= 0.1.4)
+    memcache-client (1.8.5)
     mime-types (1.17.2)
     multi_json (1.0.4)
     netrc (0.7.1)
@@ -109,6 +110,7 @@ DEPENDENCIES
   heroku
   json
   launchy
+  memcache-client
   png
   rack-test (= 0.5.6)
   rest-client
diff --git a/server/sinatra/server.rb b/server/sinatra/server.rb
index 68d0dbe..a9b098a 100644
--- a/server/sinatra/server.rb
+++ b/server/sinatra/server.rb
@@ -24,7 +24,11 @@ class Controller < Sinatra::Base
   set :views , File.join(SINATRA_ROOT, "views")
   set :haml, :format => :html5
   set :versions, `git log -10 --oneline` || "no git log"
-  enable :sessions
+  if ENV.include?('SESSION_STORE')
+    use ENV['SESSION_STORE'].split('::').inject(Object) { |mod, const| mod.const_get(const) }
+  else
+    enable :sessions
+  end
   helpers ServerHelpers
 
   Store.set ENV['STORE_TYPE'], APP_ROOT
